syntax/leaninfo.vim

```diff
@@ -16,7 +16,7 @@ hi def link leanInfoGoalHyp Type
 hi def link leanInfoGoalVDash Operator
 hi def link leanInfoExpectedType Special
 
-hi def link leanInfoError DiagnosticError
-hi def link leanInfoWarning DiagnosticWarning
-hi def link leanInfoInfo DiagnosticInfo
+hi def link leanInfoError LspDiagnosticsDefaultError
```

Comment:
This'll break highlighting on the nightlies (see below for an example)

<img width="720" alt="Screen Shot 2021-09-21 at 15 45 46" src="https://user-images.githubusercontent.com/329822/134237760-cb7a0be5-3b65-432d-9173-8e2fab5afb25.png">

 -- these highlight groups don't exist anymore in them -- I think you have to do something like what I showed here: 

```suggestion
" Old neovims (0.5) pre https://github.com/neovim/neovim/pull/15585
if luaeval('vim.diagnostic == nil')
    hi def link DiagnosticError LspDiagnosticsDefaultError
    hi def link DiagnosticWarning LspDiagnosticsDefaultWarning
    hi def link DiagnosticInformation LspDiagnosticsDefaultInformation
endif

hi def link leanInfoError LspDiagnosticsDefaultError
```

to forward port the new ones to old neovim.

Reply:
Then I was confused about these lines here: https://github.com/neovim/neovim/pull/15585/files#diff-8df5fbf5803bbea8d9e9d4d8a63f0bac3a32e1925d74990a43c6327fad2aad03R28-R48

I thought that would work for backwards compatibility.  I'll add the if.

Reply:
Interesting. That looks like you should be right, but yeah I don't get this behavior on a nightly. I'll see if I can tell why whilst also seeing if I understand the test failure you're seeing.